### PR TITLE
simplify coo2hybrid & test Modularization

### PR DESCRIPTION
**Pull request by zitongzhan**
Contains the following changes
- in the logic simplification commit, placeholder was removed. This result in a non-trivial speedup once again. 
- test factory is up there. Through a single function, it is able to test multiple operators.

### CODE REVIEW

**Comment by zitongzhan** [comment]
The above commit is to alleviate the problem caused by existence of non-dense blocks. This new commit also makes coo2hybrid's execution 100% faster.

File: pypose/sparse/sparse_block_tensor.py [hunk 1]
```diff
@@ -188,32 +189,25 @@ def coo_2_hybrid(coo, proxy):
...
-    # Compute the indices of every element of coo inside their own respective blocks.
-    b_dim_t = torch.tensor(b_dim).to(dtype=torch.int64, device=coo.device).view((2, 1))
+    # in-block indices.
+    b_dim_t = torch.tensor(b_dim, dtype=torch.int64, device=coo.device).unsqueeze(-1)
     in_block_indices = coo.indices() % b_dim_t
+    # block indices.
+    block_indices = coo.indices() // b_dim_t
+    block_seq = torch.sparse_coo_tensor(
+            proxy.indices(), torch.arange(proxy.values().shape[0]), size=proxy.shape
+        ).to_dense()  # dense is fast in indexing
```
Comment (zitongzhan) [inline pypose/sparse/sparse_block_tensor.py:200]: The usage of `dense` here is for runtime concern. Indexing operation cost at least log(n) complexity for sparse tensor, but constant time for dense tensor. Here I prefer faster time over memory consumption.

**Review by huyaoyu — COMMENTED** [review]
Please see the comments. Maybe let's setup a call sometime later.

File: pypose/sparse/sparse_block_tensor.py (hunk 1 above)
Comment (zitongzhan) [reply pypose/sparse/sparse_block_tensor.py:200]: Sounds good. Can check the shape of proxy and decide whether to convert it to dense.

Comment (zitongzhan) [reply pypose/sparse/sparse_block_tensor.py]: Resolved in commit.

**Review by huyaoyu — APPROVED** [review]
Looks good. I might need to get more insight about the design of `coo_2_hybrid` when I review the high dimensional version.

**Pull request merged**